### PR TITLE
Use imagelight from build submodule to build single image

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

This is how the latest providers are built to simplify the build process. In addition, marketplace is the default registry now instead of DockerHub since it's free for all Crossplane packages.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

`make build`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,15 +11,13 @@ on:
 env:
   # Common versions
   GO_VERSION: '1.19'
-  GOLANGCI_VERSION: 'v1.50'
-  DOCKER_BUILDX_VERSION: 'v0.9.1'
+  GOLANGCI_VERSION: 'v1.50.0'
```
Comment (hasheddan) [inline .github/workflows/ci.yml:14]: I think there may be a couple of newer releases out if you want to go ahead and bump to latest 👍🏻

**Review by hasheddan — APPROVED** [review]
Awesome, thanks @muvaf! One non-blocking comment 👍🏻

File: .github/workflows/ci.yml (hunk 1 above)
Comment (muvaf) [reply .github/workflows/ci.yml:14]: Sure. But it should be same with the one we use in Makefile https://github.com/upbound/build/pull/229

Comment (hasheddan) [reply .github/workflows/ci.yml:14]: Agreed 👍🏻 can also override in Makefile in this repo

**Pull request merged**